### PR TITLE
[W5.7][W16-3]Gary Goh Yipeng

### PR DESCRIPTION
**Pull request by GaryGohYP**

### CODE REVIEW

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 1]
```diff
@@ -81,6 +82,13 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    public List<ReadOnlyPerson> sortedListView() {
+        List<ReadOnlyPerson> newInternalList = new ArrayList<>(internalList);
+        Collections.sort(newInternalList, (ReadOnlyPerson p1, ReadOnlyPerson p2) -> p1.getName().fullName.compareToIgnoreCase(p2.getName().fullName));
```
Comment (tristonpang) [inline src/seedu/addressbook/data/person/UniquePersonList.java:87]: Looks good!

Currently only sorted by name, may want to consider adding the ability to sort by other fields like address, etc

**Review by tristonpang — COMMENTED** [review]
Code looks good!

Suggestions made for additional features

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package seedu.addressbook.commands;
+
+
+import java.util.List;
+import java.util.Collections;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+
```
Comment (joanneong) [inline src/seedu/addressbook/commands/SortCommand.java:8]: Redundant blank lines here and above as well.

Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

Comment (joanneong) [reply src/seedu/addressbook/commands/SortCommand.java:8]: Same for other areas in the code below!

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Comparator;
```
Comment (joanneong) [inline src/seedu/addressbook/data/person/UniquePersonList.java:9]: The order of this import statement violates the coding standards. Why? :P

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 1 above)
Comment (joanneong) [reply src/seedu/addressbook/data/person/UniquePersonList.java:87]: Avoid lines (of code) that are longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 3 above)
Comment (joanneong) [reply src/seedu/addressbook/data/person/UniquePersonList.java:9]: Same for some of your other import statements below!

File: test/expected.txt [hunk 4]
```diff
@@ -138,6 +140,15 @@
...
 || ===================================================
+|| Enter command: || [Command entered: sort]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 	2. Betsy Choo Tags: [secretive]
+|| 	3. Charlie Dickson Email: charlie.d@nus.edu.sg Address: 333, gamma street Tags: [school][friends]
+|| 	4. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	5. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
+|| 
+|| 5 persons listed!
+|| ===================================================
```
Comment (joanneong) [inline test/expected.txt:151]: Is this an effective test for the `sort` command? What is the state of the address book before you ran this command?

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 5]
```diff
@@ -9,18 +9,15 @@
...
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
-import seedu.addressbook.data.person.Address;
-import seedu.addressbook.data.person.Email;
-import seedu.addressbook.data.person.Name;
-import seedu.addressbook.data.person.Person;
-import seedu.addressbook.data.person.Phone;
-import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.*;
```
Comment (joanneong) [inline test/java/seedu/addressbook/data/AddressBookTest.java:20]: There is something wrong with this import statement as well. Do you know what is wrong? :(

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 6]
```diff
@@ -135,4 +132,17 @@ public void getAllPersons() throws Exception {
 
         assertTrue(isIdentical(allPersons, personsToCheck));
     }
+    @Test
+    public void getSortedPersons() throws Exception {
```
Comment (joanneong) [inline test/java/seedu/addressbook/data/AddressBookTest.java:136]: What is the proper naming format for test cases? 🤔

**Review by joanneong — APPROVED** [review]
Nice to see that you have updated your documentation and tests in addition to implementing the enhancement :)

Please close this PR after reading my comments!

**Pull request closed without merging**